docs/1-getting-started/4-making-a-blog/README.md

```diff
@@ -0,0 +1,208 @@
+# Making a Blog
+
+So far we only considered contributing pages visually, directly in Page Builder. This is great for individual, unique pages but it's not the best way to manage a blog. In a blog, you want all your articles to have the same structure and style, and you want to be able to manage them all in one place.
+
+## Blog Post Type
+
+Let's start by creating a node type called `blogPost`. We'll keep it concise but feel free to add more fields if you need them. Create a new file named `src/components/BlogPost/definition.cnd`:
+
+```cnd
+[hydrogen:blogPost] > jnt:content, mix:title, jmix:mainResource, hydrogenmix:component
+ - subtitle (string) i18n mandatory
+ - authors (string) multiple
+ - cover (weakreference, picker[type='image']) mandatory < jmix:image
+ - body (string, richtext) i18n mandatory
+```
+
+You might notice the `multiple` keyword next to authors: its allows setting a list of strings instead of a single one. Because it is not marked as mandatory, it may be undefined: we will treat this case as an empty list. `mix:title` is a mixin that adds a `jcr:title` field to the node type.
+
+Create a `types.ts` file in the same folder:
+
+```ts
+import type { JCRNodeWrapper } from "org.jahia.services.content";
+
+export type Props = {
+  "jcr:title": string;
+  "subtitle": string;
+  "authors"?: string[];
+  "cover": JCRNodeWrapper;
+  "body": string;
+};
+```
+
+Let's also create a simple view to render the blog post as a card:
+
+<details>
+<summary><code>src/components/BlogPost/default.server.tsx</code></summary>
+
+```tsx
+import { buildNodeUrl, jahiaComponent } from "@jahia/javascript-modules-library";
+import type { Props } from "./types.js";
+import classes from "./component.module.css";
+
+jahiaComponent(
+  {
+    componentType: "view",
+    nodeType: "hydrogen:blogPost",
+    displayName: "Blog Post",
+  },
+  ({ "jcr:title": title, subtitle, authors = [], cover }: Props, { currentNode }) => {
```

Comment:
> hydrogen/src/components/BlogPost/default.server.tsx
  11:46  warning  A/an 'array expression' as default prop. This could lead to potential infinite render loop in React. Use a variable instead of 'array expression'  @eslint-react/no-unstable-default-props

To remove that ESLint warning:
```suggestion
const defaultAuthors: string[] = [];

jahiaComponent(
  {
    componentType: "view",
    nodeType: "hydrogen:blogPost",
    displayName: "Blog Post",
  },
  ({ "jcr:title": title, subtitle, authors = defaultAuthors, cover }: Props, { currentNode }) => {
```

Reply:
Ah I missed it. It's unfortunate because this is not relevant (server components are not reactive). Not sure the best thing to suggest here

Reply:
why not just removing the default value and use `{authors && authors.length > 0 && <p>Written by {authors.join(", ")}</p>}`?

Reply:
Yep let's do that